aws/data_source_aws_lambda_invoke.go

```diff
@@ -0,0 +1,91 @@
+package aws
+
+import (
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsLambdaInvoke() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsLambdaInvokeRead,
+
+		Schema: map[string]*schema.Schema{
+			"function_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"qualifier": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "$LATEST",
+			},
+
+			"input": {
+				Type:     schema.TypeMap,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"result": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAwsLambdaInvokeRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lambdaconn
+
+	functionName := d.Get("function_name").(string)
+	qualifier := d.Get("qualifier").(string)
+
+	payload, err := json.Marshal(d.Get("input"))
+
+	if err != nil {
+		return err
+	}
+
+	res, err := conn.Invoke(&lambda.InvokeInput{
+		FunctionName:   aws.String(functionName),
+		InvocationType: aws.String(lambda.InvocationTypeRequestResponse),
```

Comment:
In the future, we can allow a new `invocation_type` attribute for calling Lambda functions asynchronously with `Event` 👍

Reply:
Exactly, I wanted to keep the first version simple, but I think we can pass some *random* s3 key  to lambda during invocation and expect to get result stored there.